src/validators/uuid.rs

```diff
@@ -0,0 +1,177 @@
+use pyo3::intern;
+use pyo3::prelude::*;
+use pyo3::sync::GILOnceCell;
+use pyo3::types::PyBytes;
+use pyo3::types::{PyDict, PyType};
+use uuid::Uuid;
+
+use crate::build_tools::is_strict;
+use crate::errors::{ErrorType, ValError, ValResult};
+use crate::input::Input;
+use crate::recursion_guard::RecursionGuard;
+use crate::tools::SchemaDict;
+
+use super::model::create_class;
+use super::model::force_setattr;
+use super::{BuildValidator, CombinedValidator, Definitions, DefinitionsBuilder, Extra, Validator};
+
+const UUID_INT: &str = "int";
+const UUID_IS_SAFE: &str = "is_safe";
+
+static UUID_TYPE: GILOnceCell<Py<PyType>> = GILOnceCell::new();
+
+fn import_type(py: Python, module: &str, attr: &str) -> PyResult<Py<PyType>> {
+    py.import(module)?.getattr(attr)?.extract()
+}
+
+fn get_uuid_type(py: Python) -> PyResult<&PyType> {
+    Ok(UUID_TYPE
+        .get_or_init(py, || import_type(py, "uuid", "UUID").unwrap())
+        .as_ref(py))
+}
+
+#[derive(Debug, Clone, Copy)]
+enum Version {
+    UUIDv1 = 1,
+    UUIDv3 = 3,
+    UUIDv4 = 4,
+    UUIDv5 = 5,
+}
+
+impl From<Version> for usize {
+    fn from(v: Version) -> Self {
+        v as usize
+    }
+}
+
+impl From<u8> for Version {
+    fn from(u: u8) -> Self {
+        match u {
+            1 => Version::UUIDv1,
+            3 => Version::UUIDv3,
+            4 => Version::UUIDv4,
+            5 => Version::UUIDv5,
+            _ => unreachable!(),
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct UuidValidator {
+    strict: bool,
+    version: Option<Version>,
+}
+
+impl BuildValidator for UuidValidator {
+    const EXPECTED_TYPE: &'static str = "uuid";
+
+    fn build(
+        schema: &PyDict,
+        config: Option<&PyDict>,
+        _definitions: &mut DefinitionsBuilder<CombinedValidator>,
+    ) -> PyResult<CombinedValidator> {
+        let py = schema.py();
+        let version = schema.get_as::<u8>(intern!(py, "version"))?.map(Version::from);
+        Ok(Self {
+            strict: is_strict(schema, config)?,
+            version,
+        }
+        .into())
+    }
+}
+
+impl Validator for UuidValidator {
+    fn validate<'s, 'data>(
+        &'s self,
+        py: Python<'data>,
+        input: &'data impl Input<'data>,
+        extra: &Extra,
+        _definitions: &'data Definitions<CombinedValidator>,
+        _recursion_guard: &'s mut RecursionGuard,
+    ) -> ValResult<'data, PyObject> {
+        let class = get_uuid_type(py)?;
+        if let Some(py_input) = input.input_is_instance(class) {
+            Ok(py_input.to_object(py))
+        } else if extra.strict.unwrap_or(self.strict) && input.is_python() {
+            Err(ValError::new(
+                ErrorType::UuidExactType {
+                    class_name: self.get_name().to_string(),
+                },
```

Comment:
Rather than creating a new type here, maybe let's reuse the `IsInstanceOf` error which will keep the error more similar with the existing `pydantic` implementation?

```suggestion
                ErrorType::IsInstanceOf {
                    class: class.name().unwrap_or("UUID").to_string(),
                },
```